File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.java
class Solution {
    public int[] searchRange(int[] nums, int target) {
        if(nums.length == 1 && nums[0] == target) return new int[]{0,0} ;
        if(nums.length == 0 || nums.length == 1) return new int[]{-1,-1} ;
        int left = 0; int right = nums.length-1;

        while(left < right){
            if(nums[left] == target && nums[right] == target){
                break;
            }

            else if(nums[left] == target && nums[right] != target){
                right--;
            }

            else if(nums[left] != target && nums[right] == target){
                left++;
            }

            else{
                left++;
                right--;
            }
        }

        if(nums[left]==target && nums[right]==target ){
            int[] arr = new int[2];
            arr[0] = left;
            arr[1] = right;
            return arr;
        }

        return new int[]{-1,-1};
    }
}

File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.cpp
class Solution {
public:
    vector<int> searchRange(vector<int>& nums, int target) {
        int start = -1, end = -1;
        int size = nums.size();

        for(int i=0 ; i<size ; i++) {
            if(nums[i]==target) {
                start = i;
                break;
            }
        }

        for(int i=size-1 ; i>=0 ; i--) {
            if(nums[i]==target) {
                end = i;
                break;
            }
        }

        return {start,end};
    }
};